.github/workflows/060_generate_benchmark_weights.yml

```diff
@@ -0,0 +1,73 @@
+name: Generate benchmark weights
+
+on:
+  workflow_dispatch:
+
+jobs:
+  benchmark-weights:
+    runs-on: [self-hosted, tfchainrunner01]
+    container:
+      image: threefolddev/tfchain:4
+      env:
+        DEBIAN_FRONTEND: noninteractive
+        PATH: /root/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin
+        RUSTUP_HOME: /root/.rustup
+        CARGO_HOME: /root/.cargo
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Chown user
+        run: |
+          chown -R $USER:$USER $GITHUB_WORKSPACE
+
+      - name: Cache build
+        uses: actions/cache@v3
+        timeout-minutes: 6
+        continue-on-error: true
+        with:
+          path: |
+            /root/.cargo/bin/
+            /root/.cargo/registry/index/
+            /root/.cargo/registry/cache/
+            /root/.cargo/git/db/
+            substrate-node/target/
+          key: ${{ runner.os }}-tfchain-build-cache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-tfchain-build-cache
+
+      - name: Build
+        run: |
+          cd substrate-node
+          cargo build --release --features runtime-benchmarks
+
+      - name: Remove old weight files
+        run: |
+          rm substrate-node/pallets/*/src/weights.rs
+          
+      - name: Run benchmarking script
+        run: |
+          cd substrate-node/.maintain
+          chmod +x ./benchmark.sh
+          ./benchmark.sh
+
+      - name: Rebuild
+        run: |
+          cd substrate-node
+          cargo build
+
+      - name: Copy new weight files
```

Comment:
Would it make sense to configure the workflow to automatically add and commit these files to the checked branch?

Reply:
Yes it would make sense, in case we generate the files related to the branch we want to commit on in order to be compatible with the branch code.
So, in practice, only for the main branch (development) I would say
But I was not aware it was possible to commit things during workflow